Revamp of Salesforce Connector Docs (#1430)
mulesoft/docs-connectors: salesforce/10.10/modules/ROOT/pages/salesforce-connector-config-topics.adoc

= Salesforce Connector 10.10 Additional Configuration Information - Mule 4
:page-aliases: connectors::salesforce/salesforce-connector-config-topics.adoc

> *Kim, I moved the last section, Generate an OAuth 2.0 Keystore file, from the Studio topic to this topic. It has more to do with Mule than the connector (but has some relation to the connnector), so the current intro does not work. I am not sure how to introduce these sections.*

Reviewer: Make sure you remove these comments.

Configure Anypoint Connector for Salesforce (Salesforce Connector) to:

* Load data in batches
* Specify a Lead ID for a Lead Convert Request operation

> * Update Field Values to Null

Reviewer 2: ```suggestion
* Update field values to `null`
```
These are plain-text task list items, not references to the titles (per previous peer list items).

> * Generate an OAuth 2.0 Keystore File

Reviewer 2: ```suggestion
* Generate an OAuth 2.0 keystore file
```

            </ee:message>
</ee:transform>
----
== Update Field Values to Null

> To set field values to null, use an *Update* or *Upsert* request operation and set the name of the fields you want to nullify in the *Transform Message* component's *fieldsToNull* field:

Reviewer 2: ```suggestion
To set field values to `null`, use an *Update* or *Upsert* request operation and insert the names of the fields you want to nullify in the *Transform Message* component's *fieldsToNull* field:
```

.Specify the field names you want to nullify in the *Transform Message* component *fieldsToNull* field.
image::salesforce-connector-10-set-fields-to-null.png[Set a field to null]

== Generate a Keystore File

Salesforce Connector uses the Bouncy Castle cryptographic library to load the certificate from the keystore and sign the payload for the authentication requests.

> The library has a https://nvd.nist.gov/vuln/detail/CVE-2018-5382[vulnerability (CVE-2018-5382)] that relates to the BKS-V1 keystore file type. By default, Salesforce Connector uses JKS or PKCS12 keystores. It is best not to use BKS-V1 keystore files, as documented in the https://www.kb.cert.org/vuls/id/306792[Bouncy Castle proposed solution].

Reviewer 2: "it is best not to" is kind of wishy-washy. How about "Don't use" or "Avoid using"?

To generate a JKS keystore file:

. Go to your Mule workspace and open the command prompt (for Windows) or Terminal (for Mac).

> . Type this command and press enter:

Reviewer 2: "Enter this command"?
Avoids the verb contention (press? tap? etc.), the name of the key on the keyboard, and whether or not to capitalize the key name.

+
[source]
----
keytool -genkeypair -alias salesforce-cert -keyalg RSA -keystore salesforce-cert.jks
----
+
. Enter the following information:
+
** Password for the keystore
** Your first name and last name
** Your organization unit

> ** Name of your city, state, and the two-letter code for your country

Reviewer 2: Why use "Name of" in one but not the other?

+
The system generates a Java keystore file (JKS format) that contains a private or public key pair in your workspace.
+
. Provide the file path for the keystore in your connector configuration.
+

> Type this command and press enter:

Reviewer 2: ditto Line 66
